unified_planning/test/test_substituter.py

```diff
@@ -98,6 +98,57 @@ def test_substitution(self):
             e6 = Int(1)
             s.substitute(e6, subst)
 
+    def test_shadowing(self):
```

Comment:
Add a bit of comments to explain what you are testing: this will save your life in the future :)